lib/guardian.rb

```diff
@@ -529,9 +529,14 @@ def can_see_site_contact_details?
   end
 
   def auth_token
-    if cookie = request&.cookies[Auth::DefaultCurrentUserProvider::TOKEN_COOKIE]
-      UserAuthToken.hash_token(cookie)
+    cookie_string = request&.cookies[Auth::DefaultCurrentUserProvider::TOKEN_COOKIE].presence
+    if cookie_string
+      cookie = DiscourseAuthCookie.parse(cookie_string)
+      cookie.validate!
+      UserAuthToken.hash_token(cookie.token)
     end
+  rescue DiscourseAuthCookie::InvalidCookie
+    nil
```

Comment:
By the time this is called, we've already go through the middlewares and validated the cookie, is it possible to reuse the work done there instead?

Reply:
Possible, but kinda tricky/annoying because we'd have to inject the hashed cookie in the request env after we validate it in default current user provider and then fetch it here. Also makes a unit test in `guardian_spec.rb` a little awkward. The parse operation and 
 validations are very cheap I don't think it's worth adding extra complexity (also I've moved `validate!` so it's called inside `.parse` so 1 less line here)